litellm/proxy/db/pod_lock_manager.py

```diff
@@ -0,0 +1,127 @@
+import uuid
+from datetime import datetime, timedelta, timezone
+from typing import TYPE_CHECKING, Any
+
+from litellm._logging import verbose_proxy_logger
+from litellm.constants import DEFAULT_CRON_JOB_LOCK_TTL_SECONDS
+
+if TYPE_CHECKING:
+    from litellm.proxy.utils import PrismaClient, ProxyLogging
+else:
+    PrismaClient = Any
+    ProxyLogging = Any
+
+
+class PodLockManager:
+    """
+    Manager for acquiring and releasing locks for cron jobs.
+
+    Ensures that only one pod can run a cron job at a time.
+    """
+
+    def __init__(self, cronjob_id: str):
+        self.pod_id = str(uuid.uuid4())
+        self.cronjob_id = cronjob_id
+
```

Comment:
Add  unit tests with calling prisma for this class. We should validate the locks work as expected